### DISCUSSION TITLE
filenames are truncated when uploading multiple files

### DISCUSSION THREAD

**Discussion by ddeanto** [fastapi/fastapi#6112] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post("/upload_one")
async def upload_one(file: UploadFile = File(...)):
    """
    When uploading ./dir/filename the following prints "./dir/filename" as expected.
    """

    return file.filename


@router.post("/upload_multiple")
async def upload_multiple(files: list[UploadFile]):
    """
    When uploading ./dir/filename1 and ./dir/filename2 the following unexpectedly prints
    ["filename1", "filename2"] instead of ["./dir/filename1", "./dir/filename2"].
    """
    return [file.filename for file in files]
```

### Description

The doc comments above explains it all. When an endpoint receives a list of UploadFiles, the filenames are truncated where the paths are nowhere to be found.

### Operating System

Linux

### FastAPI Version

0.88.0

### Python Version

3.11.1

**Comment by antonio-pedro99** [comment]
@ddeanto  you are calling the filename, that is why you are getting that output. 
If you want to get the desired output like "./dir/filename1", you can use the os.path.realpath() if you want the relative path of the file, or os.path.abspath() if you want the absolute path of the file.
You can read the differences of both [here](https://courses.cs.washington.edu/courses/cse140/13wi/file-interaction.html#:~:text=Absolute%20and%20Relative%20file%20paths&text=An%20absolute%20file%20path%20describes,For%20example%2C%20example_directory.)

For your case, just do

`from os import path as p`
....
`return [p.realpath(filename=file.filename) for file in files]`

I hope it helps!

**Comment by ddeanto** [comment, marked as answer]
The solution for me was to do this:

```
def test_upload_multiple():
    url = f'{base_url}/upload_multiple'

    files = [  
       ('files', (file1_path, open(file1_path, 'rb'))),
       ('files', (file2_path, open(file2_path, 'rb'))),
    ]

    response = httpx.post(url=url, files=files)
    print(f'file_paths: {response.text}')
```

instead of this

```
def test_upload_multiple():
    url = f'{base_url}/upload_multiple'

    files = [
        ('files', open(file1_path, 'rb')),
        ('files', open(file2_path, 'rb'))
    ]

    response = httpx.post(url=url, files=files)
    print(f'file_paths: {response.text}')
```

Here is the app code:
```
from fastapi import FastAPI, UploadFile

app = FastAPI()

@app.post("/upload_one")
async def upload_one(file: UploadFile):
    return file.filename

@app.post("/upload_multiple")
async def upload_multiple(files: list[UploadFile]):
    return [file.filename for file in files]
```

**Comment by antonio-pedro99** [comment]
@ddeanto you will only upload two files?

**Comment by ddeanto** [comment]
I will upload many, like 10-20, but I kept it to 2 for this toy example
